packages/ui/src/machines/authenticator/defaultServices.ts

```diff
@@ -11,6 +11,53 @@ export const defaultServices = {
     return Auth.currentAuthenticatedUser();
   },
 
+  async handleSignUp(formData): Promise<any> {
+    return Auth.signUp(formData);
+  },
```

Comment:
I knew eventually we'd have to solve some of this tech debt :(

This is similar the original design for what `signUp` (and `signIn`, etc.) should look like for customers to replace:

```
async signUp({ username, password, attributes }) {
  return Auth.signUp({ username, password, attributes });
}
```

What we're missing is for this transform to happen _before_ calling the `signUp` or `signIn` services.

Even https://github.com/aws-amplify/amplify-ui/pull/911/files#diff-7746662de6b4ab9d0242a764c3f8ee8b1c40f29db8f3480989d14f8370d5bdd8R276 is too complex – these `services` should be mostly a pass-through for customers to mutate the data as you're doing here.

My recommendation is that `invoke: { src: 'signUp' }` instead passes the normalized `data`:
> https://xstate.js.org/docs/guides/communication.html#the-invoke-property

I see why you had to introduce a new service like `handle*` – we let our services grow too complex rather than being 1-line passthroughs to `Auth.*` methods.

Reply:
Fyi I've looked into using `invoke` with `data`: this option is only applicable when you're invoking a machine. 

> `data` - (optional, used only when invoking machines) 
  [adsf](https://xstate.js.org/docs/guides/communication.html#the-invoke-property)

Reply:
Whoops!  Well, through whatever means, what I was hoping for was:

- Our `signIn` / `signUp` services are what customers override
- These services share type definitions with the underlying `Auth.*` call
- These services, when replaced, **don't** lose functionality/behavior. (`signUp` today has an allowlist)

When typing this out, I had a realization – **just because they pass `signUp` doesn't mean that replaces the entire `services.signUp` method**.

Reply:
@ericclemmons What sort of ways should we accomplish this without having the whole service overwrite `services.signUp` ?